llvm-passes/ReadAttributes/ReadAttributes.cpp

```diff
@@ -0,0 +1,69 @@
+#define DEBUG_TYPE "ReadAttributes"
+
+#include "llvm/Pass.h"
+#include "llvm/IR/Module.h"
+#include "llvm/IR/Constants.h"
+
+#ifndef NDEBUG
+#include "llvm/IR/Verifier.h"
+#include "llvm/Support/Debug.h"
+#endif
+
+/* for stat support */
+#include "llvm/ADT/Statistic.h"
+STATISTIC(NumberFunctionsToJit, "The # of functions to jit");
+
+#include <set>
+
+using namespace llvm;
+
+static const char JitAnnotation[] = "jitme";
+
+class ReadAttributes : public ModulePass {
+    private:
+        using FunctionSet = std::set<Function*>;
```

Comment:
not critical: that could be a smallptrset, see 

http://llvm.org/docs/doxygen/html/classllvm_1_1SmallPtrSet.html